src/components/SaveProjectMenuItem.tsx

```diff
@@ -0,0 +1,37 @@
+import { MenuItem, useDisclosure } from "@chakra-ui/react";
+import { useCallback, useEffect } from "react";
+import { RiDownload2Line } from "react-icons/ri";
+import { FormattedMessage } from "react-intl";
+import { useProject } from "../user-projects-hooks";
+import SaveHexDialog from "./SaveHexDialog";
+
+const SaveProjectMenuItem = () => {
+  const { projectIOState, setProjectIOState } = useProject();
+  const saveHexDisclosure = useDisclosure();
+  const handleDownloadHex = useCallback(() => {
+    saveHexDisclosure.onOpen();
+    setProjectIOState("downloading");
```

Comment:
I'd like to understand this state more clearly. When does it get reset? What does it control.

Reply:
Makes sense. Agreed to switch to two dialogs, one with info and don't show this again, one that's just progress. Maybe a toast after the actual download. I'll try to move the "IO" state into the hook.